src/Domain/CoordinateMaps/CubicScale.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/TypeTraits.hpp"
+/// \cond
+class FunctionOfTime;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace CoordMapsTimeDependent {
+/*!
+ * \ingroup CoordMapsTimeDependentGroup
+ * \brief CubicScale map defined by \f$x = a(t)*\xi+(b(t)-a(t))\xi^3/X^2\f$,
+ *  where \f$X\f$ is the outer boundary.
+ *
+ * The map scales the coordinates \f$\xi\f$ near the center by a factor
+ * \f$a(t)\f$, while the coordinates near the outer the boundary \f$X\f$,
+ * are scaled by a factor \f$b(t)\f$. Here \f$a(t)\f$ and \f$b(t)\f$ are
+ * FunctionsOfTime.
+ *
+ * Currently, only a 1-D implementation.
+ */
+class CubicScale {
+ public:
+  static constexpr size_t dim = 1;
+
+  explicit CubicScale(double outer_boundary);
+  CubicScale() = default;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 1> operator()(
+      const std::array<T, 1>& source_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 1> inverse(
+      const std::array<T, 1>& target_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  template <typename T>
+  std::array<tt::remove_cvref_wrap_t<T>, 1> frame_velocity(
+      const std::array<T, 1>& source_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> inv_jacobian(
+      const std::array<T, 1>& source_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  template <typename T>
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> jacobian(
+      const std::array<T, 1>& source_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  template <typename T>
+  tnsr::Iaa<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> hessian(
+      const std::array<T, 1>& source_coords, double time,
+      const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+      noexcept;
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& p) noexcept;  // NOLINT
+
+ private:
+  friend bool operator==(const CubicScale& lhs, const CubicScale& rhs) noexcept;
+
+  std::string f_of_t_a_ = "expansion_a";
```

Comment:
So I'm wondering about the use of strings and unordered_maps of strings for holding the `f_of_t`'s. The strings are used as identifiers for the `f_of_t`'s yeah? So instead of using strings, could we have an enum of possible names of `f_of_t`'s, and have those as keys to the maps? If we have a finite or small set of different `f_of_t`s then using an enum might be far less error prone than strings.

Reply:
This is how it is currently done in the already approved Translation map. We discussed this when it came up in that pull request -- and this was what was settled on. Of course, that does not mean it is set in stone -- so if somebody wants to update it at a later time, that's fine with me -- I am just sticking with it for consistency.